src/OpenTelemetry.Abstractions/Trace/NoopTracer.cs

```diff
@@ -44,15 +45,22 @@ internal NoopTracer()
         public ITextFormat TextFormat => new TraceContextFormat();
 
         /// <inheritdoc/>
-        public IScope WithSpan(ISpan span)
+        public IDisposable WithSpan(ISpan span)
         {
-            return NoopScope.Instance;
+            return NoopScope;
         }
 
         /// <inheritdoc/>
         public ISpanBuilder SpanBuilder(string spanName)
         {
             return new NoopSpanBuilder(spanName);
         }
+
+        private class NoopDisposable : IDisposable
```

Comment:
Maybe worth having an `internal class NoopDisposable` like the one you've created on `OpenTelemetry`?
Perhaps when the second inner class/singleton shows up would be the right time to do it.

Reply:
I agree, when will need it the second time, we should separate it. We actually already need it, but in a different package (SDK), I thought good old copy-paste is better than public API for such a small thing.